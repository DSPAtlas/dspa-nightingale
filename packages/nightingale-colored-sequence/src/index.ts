import NightingaleColoredSequence from "./nightingale-colored-sequence";
export {
  default as ColorScaleParser,
  isColorValid,
} from "./utils/ColorScaleParser";
export type { DataBlock } from "./nightingale-colored-sequence";
export default NightingaleColoredSequence;
