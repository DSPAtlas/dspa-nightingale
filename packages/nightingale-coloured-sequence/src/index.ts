import NightingaleColouredSequence from "./nightingale-coloured-sequence";
export default NightingaleColouredSequence;
