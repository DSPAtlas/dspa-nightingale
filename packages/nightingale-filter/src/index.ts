import NightingaleFilter from "./nightingale-filter";
export type { Filter } from "./nightingale-filter";
export default NightingaleFilter;

