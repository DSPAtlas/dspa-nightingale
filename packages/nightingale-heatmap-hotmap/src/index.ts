import NightingaleHeatmapHotmap from "./nightingale-heatmap-hotmap";

export default NightingaleHeatmapHotmap;
