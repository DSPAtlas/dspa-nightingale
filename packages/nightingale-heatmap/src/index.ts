import NightingaleHeatmap from "./nightingale-heatmap";
export type { HeatmapData, HeatmapPoint } from "./nightingale-heatmap";
export default NightingaleHeatmap;
