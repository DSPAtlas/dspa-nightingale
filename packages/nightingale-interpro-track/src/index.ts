import NightingaleInterproTrack from "./nightingale-interpro-track";
export type { BaseGroup, LabelGroup } from "./nightingale-interpro-track";

export default NightingaleInterproTrack;
