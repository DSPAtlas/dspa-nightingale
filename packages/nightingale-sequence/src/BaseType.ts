export type BaseType = { position: number; aa: string };
