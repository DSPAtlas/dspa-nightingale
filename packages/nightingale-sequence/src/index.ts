import NightingaleSequence from "./nightingale-sequence";
export type { SequenceBaseType } from "./nightingale-sequence";
export default NightingaleSequence;
