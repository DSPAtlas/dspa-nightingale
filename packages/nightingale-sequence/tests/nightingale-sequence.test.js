import "../src/index";

let rendered;
const sequence = "MADYDDEFDTKASDL";

describe("nightingale-sequence tests", () => {
  beforeEach(async () => {
    document.documentElement.innerHTML = `<nightingale-sequence sequence="${sequence}" length="15" width="800"></nightingale-sequence>`;
    rendered = document.querySelector("nightingale-sequence");
    await new Promise((resolve) => requestAnimationFrame(() => resolve()));
    await rendered.updateComplete;
  });

  afterEach(() => {
    document.querySelector("nightingale-sequence").remove();
  });

  test("it should display the sequence correctly", async () => {
    const firstValue = rendered.querySelector(
      ".sequence text.base:first-child"
    ).textContent;
    expect(firstValue).toBe(sequence[0]);
    const lastValue = rendered.querySelector(
      ".sequence text.base:last-child"
    ).textContent;
    expect(lastValue).toBe(sequence[sequence.length - 1]);
    expect(rendered).toMatchSnapshot();
  });

  test("it should display the sequence correctly after zoom", (done) => {
    const initialWidth = rendered
      .querySelector(".background rect")
      .getAttribute("width");
    rendered.setAttribute("display-start", "2");
    rendered.setAttribute("display-end", "4");

    window.requestAnimationFrame(() => {
      const firstValue = rendered.querySelector(
        ".sequence text.base:first-child"
      ).textContent;
      expect(firstValue).toBe(sequence[1]);
      const lastValue = rendered.querySelector(
        ".sequence text.base:last-child"
      ).textContent;
      expect(lastValue).toBe(sequence[3]);
      expect(rendered).toMatchSnapshot();

      const newWidth = rendered
        .querySelector(".background rect")
        .getAttribute("width");
      expect(Number(newWidth)).toBeGreaterThan(Number(initialWidth));
      done();
    });
  });

  test("it should display the sequence correctly after highlight", (done) => {
    rendered.setAttribute("highlight", "2:5");
    window.requestAnimationFrame(() => {
      expect(rendered).toMatchSnapshot();
      done();
    });
  });
});
