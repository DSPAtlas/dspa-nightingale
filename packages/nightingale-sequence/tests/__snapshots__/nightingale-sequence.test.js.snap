// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-sequence tests it should display the sequence correctly 1`] = `
<nightingale-sequence
  length="15"
  sequence="MADYDDEFDTKASDL"
>
  <!---->
  <div
    class=""
  >
    <svg
      height="100"
      id=""
      width="800"
    >
      <g
        class="background"
      >
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="10"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="62"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="114"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="166"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="218"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="270"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="322"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="374"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="426"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="478"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="530"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="582"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="634"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="686"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="738"
        />
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(36,0)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(208.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(468.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(728.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            15
          </text>
        </g>
      </g>
      <g
        class="sequence"
        style="opacity: 1;"
        transform="translate(0,75)"
      >
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="36"
        >
          M
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="88"
        >
          A
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="140"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="192"
        >
          Y
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="244"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="296"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="348"
        >
          E
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="400"
        >
          F
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="452"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="504"
        >
          T
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="556"
        >
          K
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="608"
        >
          A
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="660"
        >
          S
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="712"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="764"
        >
          L
        </text>
      </g>
      <g
        class="highlighted"
      />
    </svg>
  </div>
</nightingale-sequence>
`;

exports[`nightingale-sequence tests it should display the sequence correctly after highlight 1`] = `
<nightingale-sequence
  highlight="2:5"
  length="15"
  sequence="MADYDDEFDTKASDL"
>
  <!---->
  <div
    class=""
  >
    <svg
      height="100"
      id=""
      width="800"
    >
      <g
        class="background"
      >
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="10"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="62"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="114"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="166"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="218"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="270"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="322"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="374"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="426"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="478"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="530"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="582"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="634"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="52"
          x="686"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="52"
          x="738"
        />
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(36,0)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(208.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(468.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(728.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            15
          </text>
        </g>
      </g>
      <g
        class="sequence"
        style="opacity: 1;"
        transform="translate(0,75)"
      >
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="36"
        >
          M
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="88"
        >
          A
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="140"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="192"
        >
          Y
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="244"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="296"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="348"
        >
          E
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="400"
        >
          F
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="452"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="504"
        >
          T
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="556"
        >
          K
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="608"
        >
          A
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="660"
        >
          S
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="712"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="764"
        >
          L
        </text>
      </g>
      <g
        class="highlighted"
      >
        <rect
          fill="#FFEB3B66"
          height="100"
          style="pointer-events: none;"
          width="208"
          x="62"
        />
      </g>
    </svg>
  </div>
</nightingale-sequence>
`;

exports[`nightingale-sequence tests it should display the sequence correctly after zoom 1`] = `
<nightingale-sequence
  display-end="4"
  display-start="2"
  length="15"
  sequence="MADYDDEFDTKASDL"
>
  <!---->
  <div
    class=""
  >
    <svg
      height="100"
      id=""
      width="800"
    >
      <g
        class="background"
      >
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="260"
          x="10"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="100"
          style="opacity: 1;"
          width="260"
          x="270"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="100"
          style="opacity: 1;"
          width="260"
          x="530"
        />
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(140,0)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(260.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(520.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(780.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
      </g>
      <g
        class="sequence"
        style="opacity: 1;"
        transform="translate(0,75)"
      >
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="140"
        >
          A
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="400"
        >
          D
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="660"
        >
          Y
        </text>
      </g>
      <g
        class="highlighted"
      />
    </svg>
  </div>
</nightingale-sequence>
`;
